extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/session/TransactionScopedStatelessSession.java

```diff
@@ -394,6 +396,30 @@ public Object get(String entityName, Object id, LockMode lockMode) {
         }
     }
 
+    @Override
+    public Filter enableFilter(String filterName) {
```

Comment:
No rush, but it might be worth adding a base class in ORM for these use cases.

I always considered the "Session API" would be rock stable and not worth the hassle, but in practice it seems there's some adjustment needed quite often..

Reply:
See https://hibernate.atlassian.net/browse/HHH-17990. Please have a look at the comments too, as it's not as straightforward as one might think, given what Quarkus does in such "proxy" classes.

BTW if only we had support for generating proxies at extension build time, we could solve this much more easily on the Quarkus side...